syntax = "proto3";

package sentiric.llm.v1;

// Sentiric ekosistemindeki diğer servislerle uyumluluk için
option go_package = "github.com/sentiric/sentiric-contracts/gen/go/llm/v1;llmv1";

service LLMLocalService {
  rpc LocalGenerateStream(LocalGenerateStreamRequest) returns (stream LocalGenerateStreamResponse);
}

message LocalGenerateStreamRequest {
  string prompt = 1;
  // Yeni: Çıkarım kalitesini artırmak için parametreler
  optional float temperature = 2; // Varsayılan: 0.8
  optional int32 top_k = 3;       // Varsayılan: 40
  optional float top_p = 4;       // Varsayılan: 0.95
}

message LocalGenerateStreamResponse {
  string token = 1;
}